src/Microsoft.Graph.Core/Requests/DeltaResponseHandler.cs

```diff
@@ -146,43 +145,107 @@ private async Task<JObject> DiscoverChangedProperties(JObject responseItem)
+            }
+        }
+
+        /// <summary>
+        /// Adds a property with the given property name to the JsonElement object. This function is currently necessary as
+        /// <see cref="JsonElement"/> is currently readonly.
+        /// </summary>
+        /// <param name="jsonElement">The Original JsonElement to add/replace a property</param>
+        /// <param name="propertyName">The property name to use</param>
+        /// <param name="newItem">The object to be added</param>
+        /// <returns></returns>
+        private string AddOrReplacePropertyToObject(JsonElement jsonElement, string propertyName, object newItem)
+        {
+            using (MemoryStream memoryStream = new MemoryStream())
+            using (Utf8JsonWriter utf8JsonWriter = new Utf8JsonWriter(memoryStream))
+            {
+                utf8JsonWriter.WriteStartObject();//write start of object
+                bool isReplacement = false;
+                foreach (var element in jsonElement.EnumerateObject())
                 {
-                    var name = parentName + property.Name;
-                    changes.Add(name);
+                    if (element.Name.Equals(propertyName))
+                    {
+                        isReplacement = true;// we are replacing an existing property
+                        utf8JsonWriter.WritePropertyName(element.Name);//write the property name
+                        // Try to get a JsonElement so that we can write it to the stream
+                        string newJsonElement = this.serializer.SerializeObject(newItem);
+                        using (var newJsonDocument = JsonDocument.Parse(newJsonElement))
+                        {
+                            newJsonDocument.RootElement.WriteTo(utf8JsonWriter);// write the object
+                        }
+                    }
+                    else
+                    {
+                        element.WriteTo(utf8JsonWriter); // write out as is
+                    }
                 }
-                else
+
+                // The property name did not exist so we a are writing something new
+                if (!isReplacement)
                 {
-                    throw new NotImplementedException("Case is not a JObject, JArray, or JProperty");
+                    utf8JsonWriter.WritePropertyName(propertyName);//write the property name
+                    // Try to get a JsonElement so that we can write it to the stream
+                    string newJsonElement = this.serializer.SerializeObject(newItem);
+                    using (var newJsonDocument = JsonDocument.Parse(newJsonElement))
+                    {
+                        newJsonDocument.RootElement.WriteTo(utf8JsonWriter);// write the object
+                    }
                 }
+
+                utf8JsonWriter.WriteEndObject();//write end of object
+                //clean up
+                utf8JsonWriter.Flush();
```

Comment:
No need to manually Flush for UTF8JsonWriter. [Dispose](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs) already does this for you.

Flushing is funnier with [MemoryStream as it does nothing](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/IO/MemoryStream.cs#L176).

Reply:
Thanks for pointing this out